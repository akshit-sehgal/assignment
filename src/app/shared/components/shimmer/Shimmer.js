import React from 'react';
import PropTypes from 'prop-types';
import './shimmer.scss';
const Shimmer = (props) => {
    const { count, showImageHolder, showLabelsHolder } = props;
    const shimmers = [];
    for (let i = 0; i < count; i++) {
        shimmers.push(
            <div key={i} className="shimmer__loading">
                <div className="sl__wrapper">
                    {showImageHolder && <div className="sl__pic sl__item--animate"></div>}
                    {
                        showLabelsHolder &&
                        <React.Fragment>
                            <div className="sl__labels sl__item--animate"></div>
                            <div className="sl__labels sl__item--animate"></div>
                        </React.Fragment>
                    }
                </div>
            </div>
        )
    }
    return <div className="shimmers-holder">{shimmers}</div>;
}
Shimmer.defaultProps = {
    count: 1,
    showImageHolder: true,
    showLabelsHolder: true
}
Shimmer.propTypes = {
    count: PropTypes.number,
    showImageHolder: PropTypes.bool,
    showLabelsHolder: PropTypes.bool
}
export default Shimmer;