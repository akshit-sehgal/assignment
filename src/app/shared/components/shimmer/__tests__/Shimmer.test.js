import React from 'react';
import { shallow } from 'enzyme';
import Shimmer from '../Shimmer';

describe('Testing <Shimmer/>', () => {
    let props;
    beforeEach(() => {
        props = {
            count: 2,
            showImageHolder: true,
            showLabelsHolder: true
        }
    })
    it('should render the different views for different props', () => {
        const wrapper = shallow(<Shimmer {...props} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.shimmer__loading').length).toEqual(props.count);
        expect(wrapper.find('.sl__pic').length).toEqual(2);
        expect(wrapper.find('.sl__labels').length).toEqual(4);
        wrapper.setProps({
            count: 1,
            showImageHolder: false,
            showLabelsHolder: false
        });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.shimmer__loading').length).toEqual(1);
        expect(wrapper.find('.sl__pic').length).toEqual(0);
        expect(wrapper.find('.sl__labels').length).toEqual(0);
    });
});