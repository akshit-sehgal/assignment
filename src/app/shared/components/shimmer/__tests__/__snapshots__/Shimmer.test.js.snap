// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Shimmer/> should render the different views for different props 1`] = `
<div
  className="shimmers-holder"
>
  <div
    className="shimmer__loading"
    key="0"
  >
    <div
      className="sl__wrapper"
    >
      <div
        className="sl__pic sl__item--animate"
      />
      <div
        className="sl__labels sl__item--animate"
      />
      <div
        className="sl__labels sl__item--animate"
      />
    </div>
  </div>
  <div
    className="shimmer__loading"
    key="1"
  >
    <div
      className="sl__wrapper"
    >
      <div
        className="sl__pic sl__item--animate"
      />
      <div
        className="sl__labels sl__item--animate"
      />
      <div
        className="sl__labels sl__item--animate"
      />
    </div>
  </div>
</div>
`;

exports[`Testing <Shimmer/> should render the different views for different props 2`] = `
<div
  className="shimmers-holder"
>
  <div
    className="shimmer__loading"
    key="0"
  >
    <div
      className="sl__wrapper"
    />
  </div>
</div>
`;
